website/blog/2025-05-22-podman-desktop-core.md

```diff
@@ -0,0 +1,144 @@
+---
+title: Containers and Kubernetes development with Podman Desktop
+description: A step-by-step guide for your containers and Kubernetes development with Podman Desktop
+authors: [firewall]
+tags: [podman-desktop, podman, development, container, tools]
+hide_table_of_contents: false
+---
+
+In the world of modern software development, containers and Kubernetes are no longer optional; they are essential. That’s where Podman Desktop comes in, your ultimate tool for building, managing, and deploying containers and Kubernetes clusters with ease and confidence. In this blogpost we will walk through a typical development workflow, creating and building a container and then testing it in a local Kubernetes cluster. Let’s dive in.
+
+## Building Containerized Applications with Podman Desktop
+
+First, let's start by building an application in a container with Podman Desktop. We need our application code and a Containerfile. For a simple application, your Containerfile might look something like:
+
+```Dockerfile
+FROM docker.io/nginxinc/nginx-unprivileged
+
+COPY <<EOF /usr/share/nginx/html/index.html
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Simple Nginx Container</title>
+</head>
+<body>
+    <h1>Hello from my Podman Nginx Container!</h1>
+    <p>This content is being served by Nginx running on a Podman container.</p>
+</body>
+</html>
+EOF
+```
+
+We use the `nginx-unprivileged` base image as a deliberate choice for enhanced security. This non-root approach adheres to the principle of least privilege, significantly limiting potential damage if the application is compromised and ensuring better compatibility with secure production environments like Kubernetes and OpenShift.
```

Comment:
it's generic AI stuff, it should really explain the issue of nginx image

To enhance security, we utilize the `nginx-unprivileged` image. This helps avoid root access, which is enforced by default in some Kubernetes distributions like OpenShift. The default NGINX image uses port 80, which is forbidden in rootless mode.